soroban-sdk/src/testutils.rs

```diff
@@ -181,11 +181,4 @@ pub trait Address {
     /// shouldn't normally matter though, as contracts should be agnostic to
     /// the underlying Address value.
     fn random(env: &Env) -> crate::Address;
-
-    /// Get the contract ID of an Address as a BytesN<32>.
-    ///
-    /// ### Panics
-    ///
-    /// If the Address is not a Contract.
-    fn contract_id(&self) -> crate::BytesN<32>;
```

Comment:
Wait, we can't make this private as it's needed by anyone creating auths in tests doesn't it? We could replace it with a to_scaddress() function.

Reply:
No, all the invocations are using address directly, so there is no need to use the `contract_id`. The fact that we even need it at all is a consequence of some older test methods that haven't been updated. I think we should also eliminate all these gaps as well and consistently use `Address` everyweher.
Also, there is no `to_scaddress()` function, but there is the respective `try_from_val` conversion that is public.